### DISCUSSION TITLE
Option for output lines with CRLF

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#1952] [General] [answered]
I think the current behavior ripgrep uses for EOL output is to keep the EOL of the lines (CR or CRLF etc.). Please clarify my impression and if there is a difference for Linux and Windows operating systems.

Personally, I'm on Windows so I would prefer to output always CRLF.
**XSV** has got such an option:
`--crlf                     Use '\r\n' line endings in the output.  `
`--crlf` has another meaning for **rigprep** but maybe an option `--crlf-output` could be added?

**Comment by BurntSushi** [comment, marked as answer]
ripgrep will print CRLF if that's what the input contains. Otherwise, ripgrep is not a tool for converting line endings. You can use some other tool for that.

xsv has an option because CRLF is part of the CSV definition.

**Reply by datatraveller1** [reply]
Yes. I can pipe it to such a tool, e.g. todos:
`rg --field-match-separator=\t --line-number "search_string" infile_tab.txt | todos > outfile_tab.txt`

BTW, I have just discovered how wonderful the new `--field-match-separator` option is. With its help the line numbers of the found lines can be added as first new field to a tab separated output file.
